File: support_tools/swigwin-3.0.12/Examples/test-suite/octave/template_inherit_runme.m
# do not dump Octave core
if exist("crash_dumps_octave_core", "builtin")
  crash_dumps_octave_core(0);
endif

template_inherit
a = FooInt();
b = FooDouble();
c = BarInt();
d = BarDouble();
e = FooUInt();
f = BarUInt();

if (!strcmp(a.blah(),"Foo"))
    error
endif

if (!strcmp(b.blah(),"Foo"))
    error
endif

if (!strcmp(e.blah(),"Foo"))
    error
endif

if (!strcmp(c.blah(),"Bar"))
    error
endif

if (!strcmp(d.blah(),"Bar"))
    error
endif

if (!strcmp(f.blah(),"Bar"))
    error
endif

if (!strcmp(c.foomethod(),"foomethod"))
    error
endif

if (!strcmp(d.foomethod(),"foomethod"))
    error
endif

if (!strcmp(f.foomethod(),"foomethod"))
    error
endif

if (!strcmp(invoke_blah_int(a),"Foo"))
    error
endif

if (!strcmp(invoke_blah_int(c),"Bar"))
    error
endif

if (!strcmp(invoke_blah_double(b),"Foo"))
    error
endif

if (!strcmp(invoke_blah_double(d),"Bar"))
    error
endif

if (!strcmp(invoke_blah_uint(e),"Foo"))
    error
endif

if (!strcmp(invoke_blah_uint(f),"Bar"))
    error
endif

File: support_tools/swigwin-3.0.12/Examples/test-suite/go/template_inherit_runme.go
package main

import . "./template_inherit"

func main() {
	a := NewFooInt()
	b := NewFooDouble()
	c := NewBarInt()
	d := NewBarDouble()
	e := NewFooUInt()
	f := NewBarUInt()

	if a.Blah() != "Foo" {
		panic(0)
	}

	if b.Blah() != "Foo" {
		panic(0)
	}

	if e.Blah() != "Foo" {
		panic(0)
	}

	if c.Blah() != "Bar" {
		panic(0)
	}

	if d.Blah() != "Bar" {
		panic(0)
	}

	if f.Blah() != "Bar" {
		panic(0)
	}

	if c.Foomethod() != "foomethod" {
		panic(0)
	}

	if d.Foomethod() != "foomethod" {
		panic(0)
	}

	if f.Foomethod() != "foomethod" {
		panic(0)
	}

	if Invoke_blah_int(a) != "Foo" {
		panic(0)
	}

	if Invoke_blah_int(c) != "Bar" {
		panic(0)
	}

	if Invoke_blah_double(b) != "Foo" {
		panic(0)
	}

	if Invoke_blah_double(d) != "Bar" {
		panic(0)
	}

	if Invoke_blah_uint(e) != "Foo" {
		panic(0)
	}

	if Invoke_blah_uint(f) != "Bar" {
		panic(0)
	}
}
